rosbag2_to_video/api.py

```diff
@@ -89,8 +112,11 @@ def convert_bag_to_video(args):
         msg_count += 1
         _, data, _ = bag_reader.read_next()
         image_msg = deserialize_message(data, msg_type)
-        # cv2.VideoWriter expects BGR encoding 
-        cv_image = cvbridge.imgmsg_to_cv2(image_msg, 'bgr8')
+        # cv2.VideoWriter expects BGR encoding
+        if type_map[args.topic] == 'sensor_msgs/msg/Image':
+            cv_image = cvbridge.imgmsg_to_cv2(image_msg, 'bgr8')
+        elif type_map[args.topic] == 'sensor_msgs/msg/CompressedImage':
+            cv_image = cvbridge.compressed_imgmsg_to_cv2(image_msg, 'bgr8')
```

Comment:
the repetition is fine for the moment, but I will refactor the code later to avoid repetition